File: FinanceApp.Mobile/src/utils/countryList.ts
export type CountryOption = {
  code: string;
  name: string;
};

/**
 * Small country list used for profile country selection.
 * ISO 3166-1 alpha-2 codes (must match API expectation in `CountryCode`).
 */
export const COUNTRY_OPTIONS: CountryOption[] = [
  { code: '', name: '— Select country —' },
  { code: 'TZ', name: 'Tanzania' },
  { code: 'UG', name: 'Uganda' },
  { code: 'KE', name: 'Kenya' },
  { code: 'RW', name: 'Rwanda' },
  { code: 'US', name: 'United States' },
  { code: 'GB', name: 'United Kingdom' },
  { code: 'CA', name: 'Canada' },
  { code: 'AU', name: 'Australia' },
  { code: 'DE', name: 'Germany' },
  { code: 'FR', name: 'France' },
  { code: 'IN', name: 'India' },
  { code: 'ZA', name: 'South Africa' },
  { code: 'NG', name: 'Nigeria' },
  { code: 'GH', name: 'Ghana' },
  { code: 'ET', name: 'Ethiopia' },
  { code: 'CN', name: 'China' },
  { code: 'JP', name: 'Japan' },
  { code: 'BR', name: 'Brazil' },
  { code: 'MX', name: 'Mexico' },
  { code: 'NL', name: 'Netherlands' },
  { code: 'CH', name: 'Switzerland' },
  { code: 'AE', name: 'United Arab Emirates' },
  { code: 'SA', name: 'Saudi Arabia' },
  { code: 'EG', name: 'Egypt' },
  { code: 'ZW', name: 'Zimbabwe' },
  { code: 'ZM', name: 'Zambia' },
  { code: 'MW', name: 'Malawi' },
  { code: 'MZ', name: 'Mozambique' },
  { code: 'BW', name: 'Botswana' },
  { code: 'NA', name: 'Namibia' },
];

File: FinanceApp.Web/Helpers/CountryList.cs
namespace FinanceApp.Web.Helpers;

/// <summary>
/// Common countries with ISO 3166-1 alpha-2 code and display name for profile/phone.
/// </summary>
public static class CountryList
{
    /// <summary>ISO 3166-1 alpha-2 code -> Country name. Use for dropdowns and lookups.</summary>
    public static IReadOnlyList<(string Code, string Name)> All { get; } = new List<(string, string)>
    {
        ("", "— Select country —"),
        ("TZ", "Tanzania"),
        ("UG", "Uganda"),
        ("KE", "Kenya"),
        ("RW", "Rwanda"),
        ("US", "United States"),
        ("GB", "United Kingdom"),
        ("CA", "Canada"),
        ("AU", "Australia"),
        ("DE", "Germany"),
        ("FR", "France"),
        ("ES", "Spain"),
        ("IN", "India"),
        ("ZA", "South Africa"),
        ("NG", "Nigeria"),
        ("GH", "Ghana"),
        ("ET", "Ethiopia"),
        ("CN", "China"),
        ("JP", "Japan"),
        ("BR", "Brazil"),
        ("MX", "Mexico"),
        ("NL", "Netherlands"),
        ("CH", "Switzerland"),
        ("AE", "United Arab Emirates"),
        ("SA", "Saudi Arabia"),
        ("EG", "Egypt"),
        ("ZW", "Zimbabwe"),
        ("ZM", "Zambia"),
        ("MW", "Malawi"),
        ("MZ", "Mozambique"),
        ("BW", "Botswana"),
        ("NA", "Namibia"),
    };

    public static string? GetNameByCode(string? code)
    {
        if (string.IsNullOrWhiteSpace(code)) return null;
        var pair = All.FirstOrDefault(c => string.Equals(c.Code, code.Trim(), StringComparison.OrdinalIgnoreCase));
        return pair.Code == null ? null : pair.Name;
    }
}
